src/api/java/com/minecolonies/api/colony/buildings/modules/IBuildingModule.java

```diff
@@ -0,0 +1,109 @@
+package com.minecolonies.api.colony.buildings.modules;
+
+import com.minecolonies.api.colony.IColony;
+import com.minecolonies.api.colony.buildings.IBuilding;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.network.PacketBuffer;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Interface describing the different modules.
+ */
+public interface IBuildingModule
+{
+    /**
+     * Deserialize the module.
+     * @param compound the nbt compound.
+     */
+    default void deserializeNBT(CompoundNBT compound)
+    {
+
+    }
+
+    /**
+     * Serialize the module from a compound.
+     * @param compound the compound.
+     */
+    default void serializeNBT(final CompoundNBT compound)
+    {
+
+    }
+
+    /**
+     * Serialization method to send the module data to the client side.
+     * @param buf the buffer to write it to.
+     */
+    default void serializeToView(PacketBuffer buf)
+    {
+
+    }
+
+    /**
+     * On destruction hook of the building, calling into the modules.
+     */
+    default void onDestroyed()
+    {
+
+    }
+
+    /**
+     * Colony tick hook.
+     * @param colony the colony the tick is invoked from.
+     */
+    default void onColonyTick(@NotNull IColony colony)
```

Comment:
Again this has no bearing being on a default module. Some modules do not tick and as such do not need this.